dataRetriever/requestHandlers/interface.go

```diff
@@ -26,3 +33,19 @@ type HeaderRequester interface {
 	NonceRequester
 	EpochRequester
 }
+
+// ResolverRequestFactoryHandler defines the resolver requester factory handler
+type ResolverRequestFactoryHandler interface {
+	CreateResolverRequestHandler(resolverRequestArgs ResolverRequestArgs) (process.RequestHandler, error)
```

Comment:
Let's rename these a little bit: this function should be `CreateRequestHandler` as the returned type is `RequestHandler`. 
Suggestion to rename the interface to `RequestHandlerCreator` as it is shorter and better explains what it does.

Reply:
good suggestion. Renamed